engine/execution/computation/committer/committer_test.go

```diff
@@ -7,42 +7,83 @@ import (
 
-	t.Run("calls to set and prove", func(t *testing.T) {
+	// verify after committing a snapshot, proof will be generated,
+	// and changes are saved in storage snapshot
+	t.Run("CommitView should return proof and statecommitment", func(t *testing.T) {
 
-		ledger := new(ledgermock.Ledger)
-		com := committer.NewLedgerViewCommitter(ledger, trace.NewNoopTracer())
+		l := ledgermock.NewLedger(t)
+		committer := committer.NewLedgerViewCommitter(l, trace.NewNoopTracer())
 
-		var expectedStateCommitment led.State
-		copy(expectedStateCommitment[:], []byte{1, 2, 3})
-		ledger.On("Set", mock.Anything).
-			Return(expectedStateCommitment, nil, nil).
+		// CommitDelta will call ledger.Set and ledger.Prove
+
+		reg := unittest.MakeOwnerReg("key1", "val1")
+		startState := unittest.StateCommitmentFixture()
+
+		update, err := ledger.NewUpdate(ledger.State(startState), []ledger.Key{convert.RegisterIDToLedgerKey(reg.Key)}, []ledger.Value{reg.Value})
+		require.NoError(t, err)
+
+		expectedTrieUpdate, err := pathfinder.UpdateToTrieUpdate(update, complete.DefaultPathFinderVersion)
+		require.NoError(t, err)
+
+		endState := unittest.StateCommitmentFixture()
+		require.NotEqual(t, startState, endState)
+
+		// mock ledger.Set
+		l.On("Set", mock.Anything).
+			Return(ledger.State(endState), expectedTrieUpdate, nil).
 			Once()
 
-		expectedProof := led.Proof([]byte{2, 3, 4})
-		ledger.On("Prove", mock.Anything).
+			// mock ledger.Prove
+		expectedProof := ledger.Proof([]byte{2, 3, 4})
+		l.On("Prove", mock.Anything).
 			Return(expectedProof, nil).
 			Once()
 
-		newState, proof, _, err := com.CommitView(
-			&snapshot.ExecutionSnapshot{
-				WriteSet: map[flow.RegisterID]flow.RegisterValue{
-					flow.NewRegisterID("owner", "key"): []byte{1},
-				},
+			// previous block's storage snapshot
+		oldReg := unittest.MakeOwnerReg("key1", "oldvalue")
+		previousBlockSnapshot := storehouse.NewExecutingBlockSnapshot(
+			snapshot.MapStorageSnapshot{
+				oldReg.Key: oldReg.Value,
 			},
-			utils.StateCommitmentFixture())
+			flow.StateCommitment(update.State()),
+		)
+
+		// this block's register updates
+		blockUpdates := &snapshot.ExecutionSnapshot{
+			WriteSet: map[flow.RegisterID]flow.RegisterValue{
+				reg.Key: oldReg.Value,
+			},
+		}
+
+		newCommit, proof, trieUpdate, newStorageSnapshot, err := committer.CommitView(
+			blockUpdates,
+			previousBlockSnapshot,
+		)
+
 		require.NoError(t, err)
-		require.Equal(t, flow.StateCommitment(expectedStateCommitment), newState)
+
+		// verify CommitView returns expected proof and statecommitment
+		require.Equal(t, previousBlockSnapshot.Commitment(), flow.StateCommitment(trieUpdate.RootHash))
+		require.Equal(t, newCommit, newStorageSnapshot.Commitment())
+		require.Equal(t, endState, newCommit)
 		require.Equal(t, []uint8(expectedProof), proof)
+		require.True(t, expectedTrieUpdate.Equals(trieUpdate))
+
+		// TOOD(leo): verify ledger.Set and ledger.Prove are called and received expected arguments
```

Comment:
How to do this?

Reply:
You can put a function inside the mock like so:
```go
l.On("Set", mock.Anything).
			Return(func(/*args with types*/) /*return vals*/ {
                              // assert whatever you want
			     return ledger.State(endState), expectedTrieUpdate, nil
			}).
			Once()
```